sample-xml/app/src/main/java/co/nimblehq/sample/xml/ui/common/ItemDivider.kt

```diff
@@ -0,0 +1,30 @@
+package co.nimblehq.sample.xml.ui.common
+
+import android.content.Context
+import android.graphics.Canvas
+import android.view.View
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.recyclerview.widget.RecyclerView
+import co.nimblehq.sample.xml.R
+
+
+class ItemDivider(private val context: Context) : RecyclerView.ItemDecoration() {
+
+    private val divider = AppCompatResources.getDrawable(context, R.drawable.divider)
+
+    override fun onDraw(c: Canvas, parent: RecyclerView, state: RecyclerView.State) {
+        divider?.let {
+            val left = parent.paddingLeft
+            val right = parent.width - parent.paddingRight
+            val childCount = parent.childCount
+            for (i in 0 until childCount - 1) {
+                val child: View = parent.getChildAt(i)
+                val params = child.layoutParams as RecyclerView.LayoutParams
+                val top: Int = child.bottom + params.bottomMargin
+                val bottom: Int = top + it.intrinsicHeight
+                it.setBounds(left, top, right, bottom)
+                it.draw(c)
+            }
+        }
+    }
+}
```

Comment:
What about if we extend `DividerItemDecoration` instead of `RecyclerView.ItemDecoration()` then handle the `divider item` in method `getItemOffsets`, So we can get rid of `divider.xml`. Here's some idea:
```suggestion
class ItemDivider(context: Context) : DividerItemDecoration(context, VERTICAL) {

    override fun getItemOffsets(
        outRect: Rect,
        view: View,
        parent: RecyclerView,
        state: RecyclerView.State
    ) {
        val position = parent.getChildAdapterPosition(view)
        if (position == state.itemCount - 1) {
            outRect.setEmpty()
        } else {
            super.getItemOffsets(outRect, view, parent, state)
        }
    }
}
```

But this way, we will face the issue of `background` color. To use this way, we need to update the `item_home_list.xml` a bit like this:

```
<?xml version="1.0" encoding="utf-8"?>
<androidx.appcompat.widget.LinearLayoutCompat xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    android:id="@+id/lItemHome"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:background="?android:colorBackground"
    android:foreground="?android:attr/selectableItemBackground">

    ...
</androidx.appcompat.widget.LinearLayoutCompat>
```
What's do you guys think? 🙇‍♀️

Reply:
Thanks! @Wadeewee  I never thought about extending the `DividerItemDecoration` before!.

Reply:
We ended up removing the logic to remove the divider at the last item [here](https://github.com/nimblehq/android-templates/pull/281#discussion_r954672390. 🙏🏻